external-import/intel471_v2/tests/tests_connector/test_settings.py

```diff
@@ -0,0 +1,180 @@
+                    "api_key": "test-api-key",
+                    "interval_indicators": 60,
+                    "initial_history_indicators": INITIAL_HISTORY_TIMESTAMP,
+                    "interval_yara": 60,
+                    "initial_history_yara": INITIAL_HISTORY_TIMESTAMP,
+                    "interval_cves": 120,
+                    "initial_history_cves": INITIAL_HISTORY_TIMESTAMP,
+                    "interval_reports": 120,
+                    "initial_history_reports": INITIAL_HISTORY_TIMESTAMP,
+                    "proxy": None,
+                    "ioc_score": 90,
+                },
+            },
+            "opencti.url",
+            id="invalid_opencti_url",
+        ),
+        pytest.param(
+            {
+                "opencti": {
+                    "url": "http://localhost:8080",
+                    "token": "test-token",
+                },
+                "connector": {
+                    "name": "Test Connector",
+                    "scope": "test, connector",
+                    "log_level": "error",
+                },
+                "intel471": {
+                    "api_username": "test-username",
+                    "api_key": "test-api-key",
+                    "interval_indicators": 60,
+                    "initial_history_indicators": INITIAL_HISTORY_TIMESTAMP,
+                    "interval_yara": 60,
+                    "initial_history_yara": INITIAL_HISTORY_TIMESTAMP,
+                    "interval_cves": 120,
+                    "initial_history_cves": INITIAL_HISTORY_TIMESTAMP,
+                    "interval_reports": 120,
+                    "initial_history_reports": INITIAL_HISTORY_TIMESTAMP,
+                    "proxy": None,
+                    "ioc_score": 90,
+                },
+            },
+            "connector.id",
+            id="missing_connector_id",
+        ),
+    ],
+)
+def test_settings_should_raise_when_invalid_input(settings_dict, field_name):
+    """
+    Test that `ConnectorSettings` (implementation of `BaseConnectorSettings` from `connectors-sdk`) raises on invalid input.
+    For the test purpose, `BaseConnectorSettings._load_config_dict` is overridden to return
+    a fake and invalid dict (instead of the env/config vars parsed from `config.yml`, `.env` or env vars).
+
+    :param settings_dict: The dict to use as `ConnectorSettings` input
+    """
+
+    class FakeConnectorSettings(ConnectorSettings):
+        """
+        Subclass of `ConnectorSettings` (implementation of `BaseConnectorSettings`) for testing purpose.
+        It overrides `BaseConnectorSettings._load_config_dict` to return a fake but valid config dict.
+        """
+
+        @classmethod
+        def _load_config_dict(cls, _, handler) -> dict[str, Any]:
+            return handler(settings_dict)
+
+    with pytest.raises(ConfigValidationError) as err:
+        FakeConnectorSettings()
+    assert str("Error validating configuration") in str(err)
```

Comment:
You can use a regex as parameter of `pytest.raises` to ensure that a string is displayed in exception message
```suggestion
    with pytest.raises(ConfigValidationError, match="Error validating configuration") as err:
        FakeConnectorSettings()
```

Reply:
Yep that's a great improvement!
I commit your suggestion here, and I also update the migration scripts too.
Do you want to open a PR to add it to the templates (the tests of migrated connectors are copied/pasted from the templates)?